src/ImageSharp/Processing/Processors/Transforms/AffineProcessor.cs

```diff
@@ -0,0 +1,228 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.Helpers;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors
+{
+    /// <summary>
+    /// Provides the base methods to perform affine transforms on an image.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal abstract class AffineProcessor<TPixel> : ResamplingWeightedProcessor<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AffineProcessor{TPixel}"/> class.
+        /// </summary>
+        /// <param name="sampler">The sampler to perform the resize operation.</param>
+        protected AffineProcessor(IResampler sampler)
+            : base(sampler, 1, 1, Rectangles.DefaultRectangle) // Hack to prevent Guard throwing in base, we always set the canvas
```

Comment:
I think the we should not force having a know `ResizeRectangle` at constructor time. We should probably turn `CreateNewCanvas()` into a `Rectangle CalculateDestRectangle(sourceRectangle)` calling it, and setting `ResizeRectangle` inside the `BeforeApply()` step.
`ResizeProcessor` could return a static size, known at constructor time then.

Reply:
I'll have a look. The `CreateDestination` method is used to determine the size of the destination output in the base class so we require a known height/width then. This takes place before `BeforeApply` so I might not be able to do much.

Reply:
We don't use anything useful from the base class now so this will be refactored to stand alone